qos/cosmos/request_validator_rest.go

```diff
@@ -0,0 +1,246 @@
+package cosmos
+
+import (
+	"errors"
+	"net/url"
+
+	"github.com/buildwithgrove/path/gateway"
+	qosobservations "github.com/buildwithgrove/path/observation/qos"
+	"github.com/buildwithgrove/path/protocol"
+	"github.com/buildwithgrove/path/qos"
+	"github.com/buildwithgrove/path/qos/jsonrpc"
+	"github.com/pokt-network/poktroll/pkg/polylog"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+// Default timeout for REST requests
+const defaultRESTRequestTimeoutMillisec = 30000
+
+// validateRESTRequest validates a REST request by:
+// 1. Validating HTTP method and path
+// 2. Determining the specific RPC type from the path
+// 3. Checking if the RPC type is supported
+// 4. Creating the request context with all necessary information
+func (rv *requestValidator) validateRESTRequest(
+	httpRequestURL *url.URL,
+	httpRequestMethod string,
+	httpRequestBody []byte,
+) (gateway.RequestQoSContext, bool) {
+	httpRequestPath := httpRequestURL.Path
+
+	logger := rv.logger.With("validator", "REST")
+
+	// Determine the specific RPC type based on path patterns - use existing function
+	rpcType := determineRESTRPCType(httpRequestPath)
+
+	logger = logger.With(
+		"detected_rpc_type", rpcType.String(),
+		"request_path", httpRequestPath,
+	)
+
+	// Check if this RPC type is supported by the service
+	if _, supported := rv.supportedAPIs[rpcType]; !supported {
+		logger.Warn().Msg("Request uses unsupported RPC type")
+		// TODO_TECHDEBT(@adshmh): Review error context creation for REST requests.
```

Comment:
WDYT about using `-1` as the ID field for error responses for CometBFT requests? 

This seems to be the standard they use, eg. all responses to a `/status` REST endpoint return a JSON-RPC response with ID = `-1`.